src/west/app/project.py

```diff
@@ -924,6 +924,30 @@ def fetch_strategy(self, args):
         else:
             return 'smart'
 
+    def update_submodules(self, project):
+        # Updates given project submodules by using
+        # 'git submodule update --init --checkout --recursive' command
+        # from the project.path location.
+        submodules = project.submodules
+        call_location = os.path.join(self.topdir, project.path)
+
+        if submodules is None:
+            return
```

Comment:
Optimization nit: you could do this earlier. In fact you can probably do it before even calling this method which would then become the `update_submodules_list()` method (no need actually to change the name, just to get the point across).

Reply:
I moved it a bit higher, but I'm not sure about moving it outside of the method, as I would like to make sure that nobody will not use if in some other place without checking if submodules are not None.

Reply:
I failed to parse the latest sentence so I'm not sure about the concern. However this check is the very first statement now so it will be trivial to move out if needed or wanted later for any reason. So I'm happy.